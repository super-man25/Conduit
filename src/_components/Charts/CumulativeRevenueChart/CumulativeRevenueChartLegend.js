import React, { Fragment } from 'react';
import { ChartLegendItem } from '_components';
import { cssConstants } from '_constants';

export function CumulativeRevenueChartLegend() {
  return (
    <Fragment>
      <ChartLegendItem label="Actual Revenue" />
      <ChartLegendItem
        dashed
        color={cssConstants.PRIMARY_LIGHT_BLUE}
        label="Projected Revenue"
      />
    </Fragment>
  );
}
