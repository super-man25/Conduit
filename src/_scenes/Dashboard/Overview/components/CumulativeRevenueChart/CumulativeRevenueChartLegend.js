import React, { Fragment } from 'react';
import { ChartLegendItem } from '_components';
import { cssConstants } from '_constants';

export function CumulativeRevenueChartLegend({ hasProjected }) {
  return (
    <Fragment>
      <ChartLegendItem label={hasProjected ? 'Actual Revenue' : 'Revenue'} />
      {hasProjected && (
        <ChartLegendItem
          dashed
          color={cssConstants.PRIMARY_LIGHT_BLUE}
          label="Projected Revenue"
        />
      )}
    </Fragment>
  );
}
