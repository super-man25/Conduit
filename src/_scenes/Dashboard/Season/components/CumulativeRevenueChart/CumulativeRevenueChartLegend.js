import React, { Fragment } from 'react';
import { cssConstants } from '_constants';
import { ChartLegendItem } from '_components';

export function CumulativeRevenueChartLegend() {
  return (
    <Fragment>
      <ChartLegendItem label="Actual Revenue" />
      <ChartLegendItem
        dashed
        color={cssConstants.PRIMARY_LIGHT_BLUE}
        label="Projected Revenue"
      />
    </Fragment>
  );
}
