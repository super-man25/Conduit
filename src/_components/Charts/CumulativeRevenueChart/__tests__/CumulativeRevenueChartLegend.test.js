import React from 'react';
import { shallow } from 'enzyme';
import { CumulativeRevenueChartLegend } from '../CumulativeRevenueChartLegend';

describe('<CumulativeRevenueChartLegend />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<CumulativeRevenueChartLegend />);
    expect(wrapper).toMatchSnapshot();
  });
});
