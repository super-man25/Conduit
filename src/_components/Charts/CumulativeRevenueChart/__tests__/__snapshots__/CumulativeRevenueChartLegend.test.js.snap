// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CumulativeRevenueChartLegend /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CumulativeRevenueChartLegend />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Unknown
          color="#38A9DB"
          label="Actual Revenue"
/>,
        <Unknown
          color="#38A9DB"
          dashed={true}
          label="Projected Revenue"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "#38A9DB",
          "label": "Actual Revenue",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "#38A9DB",
          "dashed": true,
          "label": "Projected Revenue",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Unknown
            color="#38A9DB"
            label="Actual Revenue"
/>,
          <Unknown
            color="#38A9DB"
            dashed={true}
            label="Projected Revenue"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "#38A9DB",
            "label": "Actual Revenue",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "#38A9DB",
            "dashed": true,
            "label": "Projected Revenue",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
